honeybee/model.py

```diff
@@ -1349,6 +1351,46 @@ def to_dict(self, included_prop=None, triangulate_sub_faces=False):
 
         return base
 
+    def to_hbjson(self, folder_path=None, name="unnamed", indent=None, included_prop=None, triangulate_sub_faces=False):
+        """Writes a Honeybee model to HBJSON.
+
+        Args:
+            folder_path: A text string of path to folder where HBJSOn will be written. Defaults to None.
+            name: A text string that will be the name of the HBJSOn. Defaults to "unnamed".
+            indent: A positive integer to set the indentation used in the
+                resulting HBJSON file. If None or 0, the JSON will be a single line. Defaults to None.
+            included_prop: List of properties to filter keys that must be included in
+                output dictionary. For example ['energy'] will include 'energy' key if
+                available in properties to_dict. By default all the keys will be
+                included. To exclude all the keys from extensions use an empty list.
+            triangulate_sub_faces: Boolean to note whether sub-faces (including
+                Apertures and Doors) should be triangulated if they have more than
+                4 sides (True) or whether they should be left as they are (False).
+                This triangulation is necessary when exporting directly to EnergyPlus
+                since it cannot accept sub-faces with more than 4 vertices. Note that
+                setting this to True will only triangulate sub-faces with parent Faces
+                that also have parent Rooms since orphaned Apertures and Faces are
+                not relevant for energy simulation. Default: False.
+        """
+        # Create dictionary from the Honeybee Model
+        hb_dict = self.to_dict(included_prop=None, triangulate_sub_faces=False)
+
+        # Setting up a name for the HBJSON
+        if not name.lower().endswith('.hbjson'):
+            file_name = '{}.hbjson'.format(name)
+        else:
+            name = name[:-7]
```

Comment:
I'm not sure why this is needed here?

Reply:
This is in case a user gives a file name of ,"testhbjson".

Reply:
Hmm... why don't you just use `test.hbjson`? You are removing the extension in this line and then adding it back again in next line?

Reply:
I get your point. I am removing `.hbjson` from the name and adding back in. Whereas I could just use that as a name. I think I will change that.